runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkJobServerDriver.java

```diff
@@ -239,7 +239,7 @@ private InMemoryJobService createJobService() throws IOException {
       throws IOException {
     BeamFileSystemArtifactStagingService service = new BeamFileSystemArtifactStagingService();
     final GrpcFnServer<BeamFileSystemArtifactStagingService> artifactStagingService;
-    if (Strings.isNullOrEmpty(configuration.host)) {
+    if (configuration.artifactPort == 0) {
```

Comment:
Should this be `Strings.isNullOrEmpty(configuration.host) || configuration.port == 0`?